utils/fragmented_temporary_buffer.hh

```diff
@@ -60,6 +60,13 @@ public:
         return ostream::simple(reinterpret_cast<char*>(current.get_write()), current.size());
     }
 
+    auto begin() const {
```

Comment:
Please use explicit types, an API shouldn't require a reader to dig into the implementation.